src/main/java/org/lsc/plugins/connectors/james/JamesUserDstService.java

```diff
@@ -0,0 +1,137 @@
+
+    private final JamesDao jamesDao;
+    private final JamesService service;
+    private final Class<IBean> beanClass;
+
+    public JamesUserDstService(final TaskType task) throws LscServiceConfigurationException {
+        try {
+            if (task.getPluginDestinationService().getAny() == null
+                || task.getPluginDestinationService().getAny().size() != 1
+                || !((task.getPluginDestinationService().getAny().get(0) instanceof JamesUsersService))) {
+                throw new LscServiceConfigurationException("Unable to identify the James service configuration " + "inside the plugin source node of the task: " + task.getName());
+            }
+
+            this.service = (JamesService) task.getPluginDestinationService().getAny().get(0);
+            this.beanClass = (Class<IBean>) Class.forName(task.getBean());
+            PluginConnectionType connection = (PluginConnectionType) service.getConnection().getReference();
+            this.jamesDao = new JamesDao(connection.getUrl(), connection.getPassword(), task);
+        } catch (ClassNotFoundException e) {
+            throw new LscServiceConfigurationException(e);
+        }
+    }
+
+    @Override
+    public boolean apply(LscModifications lscModifications) throws LscServiceException {
+        if (lscModifications.getMainIdentifier() == null) {
+            LOGGER.error("MainIdentifier is needed to update");
+            return false;
+        }
+        User user = new User(lscModifications.getMainIdentifier());
+        try {
+            switch (lscModifications.getOperation()) {
+                case CREATE_OBJECT:
+                    return jamesDao.addUser(user);
+                case DELETE_OBJECT:
+                    return jamesDao.removeUser(user);
+                default:
+                    LOGGER.debug("{}operation, ignored.", lscModifications.getOperation());
+                    return true;
+            }
+        } catch (ProcessingException exception) {
+            LOGGER.error(String.format("ProcessingException while writing (%s)", exception));
+            LOGGER.debug(exception.toString(), exception);
+            return false;
+        }
+    }
+
+    @Override
+    public List<String> getWriteDatasetIds() {
+        return service.getWritableAttributes().getString();
+    }
+
+    @Override
+    public IBean getBean(String pivotName, LscDatasets pivotAttributes, boolean fromSameService) throws LscServiceException {
+        LOGGER.debug(String.format("Call to getBean(%s, %s, %b)", pivotName, pivotAttributes, fromSameService));
+        if (pivotAttributes.getAttributesNames().size() < 1) {
+            return null;
+        }
+        String pivotAttribute = pivotAttributes.getAttributesNames().get(0);
+        String email = pivotAttributes.getStringValueAttribute(pivotAttribute);
+        if (email == null) {
+            return null;
+        }
+        try {
+            User user = jamesDao.isExistUser(email);
```

Comment:
Do we really need this read on James for each user?

My feeling is that we likely do not need it: the email is enough to create the user...

Reply:
When debugging old code, I realized with N email in LDAP, we need to call Jame with N time for checking.  
`getBean` has been called before `get ListPivosts`. 
I think on it pros of this way avoid do a big request (retrieve all user).
Anyway, this is a selection. We can store all users in memory (static collection) to less call API.  
What do you think?